ddl/table.go

```diff
@@ -496,3 +522,37 @@ func checkAddPartitionValue(meta *model.TableInfo, part *model.PartitionInfo) er
 	}
 	return nil
 }
+
+// doCreateTableInsert inserts data into created table from 'select' query for 'create table ... select' syntax.
+// a global session is used because this called in DDL owner server(rather than the server communicating with the client)
+// TODO: copy & set necessary session variables from original TiDB server
+func doCreateTableInsert(d *ddlCtx, t *meta.Meta, job *model.Job, tbInfo *model.TableInfo, snapshotTS uint64) error {
+	sctx, err := util.CreateSessionContext(d.store)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	closable := sctx.(interface{ Close() })
+	if closable == nil {
+		return errors.Errorf("temporary session cannot be closed, should never happen")
+	}
+	defer closable.Close()
+
+	dbInfo, err := t.GetDatabase(job.SchemaID)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	sctx.GetSessionVars().CreateTableInsertingID = tbInfo.ID
+	sctx.GetSessionVars().CurrentDB = dbInfo.Name.L
+	sctx.GetSessionVars().SnapshotTS = snapshotTS
+
+	_, err = sctx.(sqlexec.SQLExecutor).Execute(context.Background(), job.Query)
+	if err != nil {
+		job.State = model.JobStateCancelled
+		startKey := tablecodec.EncodeTablePrefix(tbInfo.ID)
+		job.Args = append(job.Args, startKey, getPartitionIDs(tbInfo))
```

Comment:
The job.Args you decode in insertJobIntoDeleteRangeTable is a tableInfo, why you push startkey and partitionIDs here?

Reply:
That was about a deprecated implementation, I just forgot to remove those codes. Thanks a lot.